src/hooks/useDebouncedCallback.ts

```diff
@@ -0,0 +1,30 @@
+import { useRef, useCallback } from "react";
+
+export const useDebouncedCallback = <T extends (...args: any[]) => void>(
+  func: T,
+  delay: number,
+) => {
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
```

Comment:
this shouldnt use types from node js runtime, as it primarily runs in the browser environment

Reply:
```suggestion
  const timeoutRef = useRef<ReturnType<typeof setTimeout>>(null);
```
simplest way to deal with it is to infer type, which will be correct regardless if this runs in browser or node

Reply:
True, changed